src/v/s3/tests/s3_client_test.cc

```diff
@@ -447,25 +447,28 @@ configured_server_and_client_pool started_pool_and_server(
     };
 }
 
+static ss::future<> test_client_pool_payload(
+  ss::shared_ptr<ss::httpd::http_server_control> server,
+  s3::client_pool::http_client_ptr client) {
+    iobuf payload;
+    auto payload_stream = make_iobuf_ref_output_stream(payload);
+    auto http_response = co_await client->get_object(
+      s3::bucket_name("test-bucket"), s3::object_key("test"), 100ms);
+    auto input_stream = http_response->as_input_stream();
+    co_await ss::copy(input_stream, payload_stream);
+    iobuf_parser p(std::move(payload));
+    auto actual_payload = p.read_string(p.bytes_left());
+    BOOST_REQUIRE_EQUAL(actual_payload, expected_payload);
+}
+
 void test_client_pool(s3::client_pool_overdraft_policy policy) {
     auto conf = transport_configuration();
     auto [server, pool] = started_pool_and_server(2, policy, conf);
     std::vector<ss::future<>> fut;
     for (size_t i = 0; i < 20; i++) {
         auto f = pool->acquire().then(
-          [_server = server](
-            s3::client_pool::client_lease lease) -> ss::future<> {
-              auto server = _server;
-              auto client = lease.client;
-              iobuf payload;
-              auto payload_stream = make_iobuf_ref_output_stream(payload);
-              auto http_response = co_await client->get_object(
-                s3::bucket_name("test-bucket"), s3::object_key("test"), 100ms);
-              auto input_stream = http_response->as_input_stream();
-              co_await ss::copy(input_stream, payload_stream);
-              iobuf_parser p(std::move(payload));
-              auto actual_payload = p.read_string(p.bytes_left());
-              BOOST_REQUIRE_EQUAL(actual_payload, expected_payload);
+          [server = server](s3::client_pool::client_lease lease) {
```

Comment:
nit:
```suggestion
          [server](s3::client_pool::client_lease lease) {
```

Reply:
server in the parent scope is not a variable and can't be captured.

Reply:
Oh yeah. Clang doesn't support this yet.